app/models/service_plan.rb

```diff
@@ -10,4 +11,45 @@ class ServicePlan < ApplicationRecord
   has_many   :service_instances
 
   acts_as_taggable
+
+  def order(catalog_id, additional_parameters)
+    payload = external_payload(catalog_id, additional_parameters)
+
+    response = service_catalog_client.order_service_plan(payload)
+
+    parsed_data = JSON.parse(response.body)
+    task = Task.create!(:tenant => tenant, :context => parsed_data['metadata']['selfLink'])
+    task.id
+  end
+
+  private
+
+  def external_payload(catalog_id, parameters)
```

Comment:
This would be better in the service_catalog_client module IMO since it is very kube specific, you can just pass in the params to `#order_service_plan`